Panic happends when deprecating metrics

**What happened**:
CI failed when deprecated metrics. Refer to https://github.com/kubernetes/kubernetes/pull/83836.

The reason is that __a hidden metrics won't be created thus can't be registered to prometheus's registry__.

I guess the other PR failed for the same reason.
#83837 #83838 #83839 #83841
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
When starting the cluster, panic happens
```
[root@ecs-d8b6 ~]# cat /tmp/kube-apiserver.log 
W1014 09:16:25.330658   31301 metric.go:103] This metric has been deprecated for more than one release, hiding.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10f827b]

goroutine 11 [running]:
k8s.io/kubernetes/vendor/k8s.io/component-base/metrics.(*HistogramVec).Describe(0xc0002c2c40, 0xc000097ce0)
<autogenerated>:1 +0xb
k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Register.func1(0x7f6d53d79100, 0xc0002c2c40, 0xc000097ce0)
/root/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus/registry.go:273 +0x3b
created by k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Register
/root/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/prometheus/client_golang/prometheus/registry.go:272 +0x154
[root@ecs-d8b6 ~]#
```
**Anything else we need to know?**:
In my option, we need to remove backoff from legacyregistry.

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/assign @logicalhan @RainbowMango 
/sig instrumentation

--- a/staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go
+++ b/staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go
@@ -49,18 +49,12 @@ func Handler() http.Handler {
 // Register registers a collectable metric but uses the global registry
 func Register(c metrics.Registerable) error {
 	err := defaultRegistry.Register(c)
-	// sideload global prom registry as fallback
-	prometheus.Register(c)
 	return err
 }
 
 // MustRegister registers registerable metrics but uses the global registry.
 func MustRegister(cs ...metrics.Registerable) {
 	defaultRegistry.MustRegister(cs...)
-	// sideload global prom registry as fallback
-	for _, c := range cs {
-		prometheus.Register(c)
-	}
 }
 
 // RawMustRegister registers prometheus collectors but uses the global registry, this
@@ -69,10 +63,6 @@ func MustRegister(cs ...metrics.Registerable) {
 // Deprecated
 func RawMustRegister(cs ...prometheus.Collector) {
 	defaultRegistry.RawMustRegister(cs...)
-	// sideload global prom registry as fallback
-	for _, c := range cs {
-		prometheus.Register(c)
-	}
 }
 
 // RawRegister registers a prometheus collector but uses the global registry, this
@@ -81,7 +71,5 @@ func RawMustRegister(cs ...prometheus.Collector) {
 // Deprecated
 func RawRegister(c prometheus.Collector) error {
 	err := defaultRegistry.RawRegister(c)
-	// sideload global prom registry as fallback
-	prometheus.Register(c)
 	return err
 }

